packages/sirius-web/frontend/sirius-web-application/src/modals/duplicate-object/DuplicateObjectModal.tsx

```diff
@@ -0,0 +1,270 @@
+
+  const [state, setState] = useState<DuplicateObjectModalState>({
+    containerSelection: { entries: [] },
+    containmentFeatureNames: [],
+    containmentFeatureName: '',
+    duplicateContent: true,
+    copyOutgoingReferences: true,
+    updateIncomingReferences: false,
+    filterBarText: '',
+    expanded: [],
+    maxDepth: 1,
+  });
+
+  const { getContainmentFeatureNames, containmentFeatureNames } = useContainmentFeatureNames();
+
+  useEffect(() => {
+    if (state.containerSelection.entries[0]) {
+      getContainmentFeatureNames({
+        variables: {
+          editingContextId,
+          containerId: state.containerSelection.entries[0].id,
+          containedObjectId: objectToDuplicateId,
+        },
+      });
+    }
+  }, [state.containerSelection, objectToDuplicateId]);
+
+  useEffect(() => {
+    if (containmentFeatureNames) {
+      setState((prevState) => ({
+        ...prevState,
+        containmentFeatureNames,
+        containmentFeatureName: containmentFeatureNames[0] ?? '',
+      }));
+    }
+  }, [containmentFeatureNames]);
+
+  const { duplicateObject, duplicatedObject } = useDuplicateObject();
+  const onDuplicate = () => {
+    duplicateObject(
+      editingContextId,
+      objectToDuplicateId,
+      state.containerSelection.entries[0]?.id,
+      state.containmentFeatureName,
+      state.duplicateContent,
+      state.copyOutgoingReferences,
+      state.updateIncomingReferences
+    );
+  };
+  useEffect(() => {
+    if (duplicatedObject) {
+      onObjectDuplicated({ entries: [duplicatedObject] });
+    }
+  }, [duplicatedObject, onObjectDuplicated]);
+
+  const onFeatureNameChange = (event: SelectChangeEvent<string>) => {
+    const value = event.target.value;
+    setState((prevState) => ({ ...prevState, containmentFeatureName: value }));
+  };
+
+  const onIsDuplicateContentChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.checked;
+    setState((prevState) => ({ ...prevState, duplicateContent: value }));
+  };
+  const onIsCopyOutgoingReferencesContentChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.checked;
+    setState((prevState) => ({ ...prevState, copyOutgoingReferences: value }));
+  };
+  const onIsUpdateIncomingReferencesChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.checked;
+    setState((prevState) => ({ ...prevState, updateIncomingReferences: value }));
+  };
+
+  const onBrowserSelection = (selection: Selection) => {
+    if (selection.entries[0]?.kind !== 'siriusWeb://document')
+      setState((prevState) => ({ ...prevState, containerSelection: selection }));
+  };
+
+  const onExpandedElementChange = (expanded: string[], maxDepth: number) => {
+    setState((prevState) => ({ ...prevState, expanded, maxDepth }));
+  };
+
+  const treeId: string = `modelBrowser://container?ownerKind=${encodeURIComponent(
```

Comment:
AHAH! There it is! Could you explain me why this feature of the explorer is based on reference widget with this tiny but critical coupling to `modelbrower://`? Preventing coupling is our number one priority.

Reply:
I'm not technically against the model browser becoming something generic but I am missing a first PR to extract the model browser out of the reference widget in some dedicated modules backend and frontend like the selection dialog.

And we would have to remove anything specific from the reference widget out of the model browser.